library/include/assert.h

```diff
@@ -0,0 +1,34 @@
+/*
+https://github.com/peterix/dfhack
+Copyright (c) 2009-2012 Petr Mrázek (peterix@gmail.com)
+
+This software is provided 'as-is', without any express or implied
+warranty. In no event will the authors be held liable for any
+damages arising from the use of this software.
+
+Permission is granted to anyone to use this software for any
+purpose, including commercial applications, and to alter it and
+redistribute it freely, subject to the following restrictions:
+
+1. The origin of this software must not be misrepresented; you must
+not claim that you wrote the original software. If you use this
+software in a product, an acknowledgment in the product documentation
+would be appreciated but is not required.
+
+2. Altered source versions must be plainly marked as such, and
+must not be misrepresented as being the original software.
+
+3. This notice may not be removed or altered from any source
+distribution.
+*/
+
+// no pragma once
+
+#include "Error.h"
+
+#undef assert
+#ifdef DFHACK_ASSERT
```

Comment:
Oh, right, I had that mixed up. I thought Release was the only different one, though (i.e. Debug and RelWithDebInfo share the NDEBUG setting), although I could be wrong about that. 

I don't *think* NDEBUG affects anything besides assert() (see also [here](https://stackoverflow.com/questions/5473556/what-is-the-ndebug-preprocessor-macro-used-for-on-different-platforms)). If you ask me, `-g` is more likely what could break ABI compatibility (and the Windows equivalent does, IIRC).